### PR TITLE
integration test checking error handling

### PR DESCRIPTION
**Pull request by raahi-chada2004**

### CODE REVIEW

File: integration_test.go [hunk 1]
```diff
@@ -344,17 +569,20 @@ func generateData(numRows int) error {
 	}
 
 	//send json marshalled data
-	for i := 0; i < numRows; i++ {
+	for i := 1; i <= numRows; i++ {
 		curr := log_entry{
 			Message: "hello world",
 		}
 		entry, err := json.Marshal(curr)
+		if sendBadRow && ((i % 5) == 0) {
```
Comment (agrawal-siddharth) [inline integration_test.go:579]: What is this magic number 5?

**Review by agrawal-siddharth — CHANGES_REQUESTED** [review]

File: integration_test.go (hunk 1 above)
Comment (raahi-chada2004) [reply integration_test.go:579]: I wanted to send 2 bad rows (one in the middle of a batch and one at the end) so I chose 5. Do you want me to change this to a constant? Maybe numRows/2?

Comment (raahi-chada2004) [reply integration_test.go:579]: Done.

File: integration_test.go [hunk 2]
```diff
@@ -33,7 +33,8 @@ const (
 	logFileName    = "logfile.log"
 	logFilePath    = "./" + logFileName
 	configFilePath = "./fluent-bit.conf"
-	numRowsData    = 10
+	numRows        = 10
+	numGoodRows    = 8
```
Comment (agrawal-siddharth) [inline integration_test.go]: Both of these constants are only used in specific tests. Better to define them as constants that are local to the scope of each of those tests.

**Review by agrawal-siddharth — CHANGES_REQUESTED** [review]

File: integration_test.go (hunk 2 above)
Comment (raahi-chada2004) [reply integration_test.go]: Done. I kept numRows a global constant as it is used in all 4 tests. I moved numGoodRows to be a local constant in the two relevant tests.

**Review by agrawal-siddharth — APPROVED** [review]

**Pull request merged**